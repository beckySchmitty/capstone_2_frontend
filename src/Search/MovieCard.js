import React from "react"

const MovieCard = ({yr, id, imdb_id, title, rating}) => {

    return (
        <div 
        className="Search-child"
        style={{
            height: `300px`,
            width: `300px`,
            backgroundColor: `lightblue`,
            color: `white`
          }}>
              <h4>{title}</h4>
              <p>{yr} {rating} {id} {imdb_id}</p>
        </div>
    )
}

export default MovieCard;