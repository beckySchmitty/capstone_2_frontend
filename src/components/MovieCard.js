import React, {useState, useEffect} from "react"
import { useSelector, useDispatch } from "react-redux";
import { useHistory } from "react-router-dom";
import { Button } from 'reactstrap';
import {getMovieFromOMDB } from "../actions/movies"


const MovieCard = ({yr, imdb_id, title, rating, src}) => {
    const history = useHistory();
    const dispatch = useDispatch();

    // Get IMDB data from backend API
    useEffect(function loadDetailsFromID() {
      async function getMovieDeets() {
        dispatch(getMovieFromOMDB(imdb_id));
      }
      getMovieDeets()
    }, [dispatch, imdb_id]);


    const toMovieDetails = (imdb_id) => {
        history.push(`/movies/${imdb_id}`);
      }

    return (
        <div 
        className="Search-MovieCard">
            <h4>{title} ({yr})</h4>
            <h6>Bechdel Rating: {rating}</h6>
            <hr></hr>
            <img style={{width: `300px`, height: `auto`}} 
            src={src}>
            </img>
            <p>IMDB_ID: {imdb_id}</p>
            <Button outline color="info" onClick={()=> toMovieDetails(imdb_id)}>More</Button>{' '}
        </div>
    )
}

export default MovieCard;