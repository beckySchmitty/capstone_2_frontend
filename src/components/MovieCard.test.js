import React from "react";
import { render } from "@testing-library/react";
import MovieCard from "./MovieCard";

// smoke test
it("renders without crashing", function() {
    render(<MovieCard />);
  });
  

// snapshot test
it("matches snapshot", function() {
  const {asFragment} = render(<MovieCard />);
  expect(asFragment()).toMatchSnapshot();
});

