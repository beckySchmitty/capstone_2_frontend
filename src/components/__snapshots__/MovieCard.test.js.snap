// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MovieCard-flex-item card card-body"
  >
    <div
      class="CardTitle-div"
    >
      <h4>
         ()
      </h4>
    </div>
    <h5
      class="CardText card-text"
    >
      Bechdel Rating: 
    </h5>
    <button
      class="MovieCard-btn"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-plus fa-w-14 fa-5x faPlus"
        data-icon="plus"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;
